### PR TITLE
Getting data for statistic from endpoint

### PR DESCRIPTION
**Pull request by ViktoriiaNakoryk**
added events and methods for getting data from endpoint

### CODE REVIEW

File: src/services/refact.ts [hunk 1]
```diff
@@ -154,6 +155,37 @@ export async function getCaps(lspUrl?: string): Promise<CapsResponse> {
   return json;
 }
 
+export function isStatisticDataResponse(
+  json: unknown,
+): json is { data: string } {
+  if (!json || typeof json !== "object") return false;
+  return typeof (json as { data?: unknown }).data === "string";
```
Comment (MarcMcIntosh) [inline src/services/refact.ts:162]: can remove the type casting with this
```suggestion
  if ( !("data" in json) ) return false
  return typeof data.json === "string";
```

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]
Looks good to me, one small suggestion and I also need to run it.
Which branch should i used for running it?

**Comment by MarcMcIntosh** [comment]
I got a 500 response code when running this because the database was not ready,
Screenshot 2024-02-26 at 16 01 18

but the spinner keeps spinning, maybe we should notify the user somehow?

File: src/hooks/useEventBusForStatistic.ts [hunk 2]
```diff
@@ -1,16 +1,140 @@
...
+      return {
+        ...state,
+        isLoading: true,
+        error: "",
+      };
+    case EVENT_NAMES_TO_STATISTIC.RECEIVE_STATISTIC_DATA:
+      return {
+        ...state,
+        statisticData: action.payload
+          ? (action.payload as StatisticData)
```
Comment (MarcMcIntosh) [inline src/hooks/useEventBusForStatistic.ts:43]: if you use an `if`s  instead of a switch you can use type guards instead of casting :)

**Review by MarcMcIntosh — APPROVED** [review]

File: src/hooks/useEventBusForStatistic.ts (hunk 2 above)
Comment (ViktoriiaNakoryk) [reply src/hooks/useEventBusForStatistic.ts]: done

**Pull request merged**